In [ ]:
pip install sentence-transformers

In [10]:
import json
import os
import pandas as pd
import requests
import re
from bs4 import BeautifulSoup
from tqdm import tqdm

# Data

Importing Andrew Tate quotes.

In [11]:
DIR = '/content/drive/MyDrive/Semantic Similarity/'
# load json data
with open(DIR + '539_2022-11-30.json') as f:
    data = json.load(f)
with open(DIR + '39.json') as g:
    data_39 = json.load(g)

data = data + data_39

Manually added tenets, ChatGPT assisted :D

In [ ]:
tenets = [
    "I believe that men have the divine imperative to become as capable, powerful and competent as possible in this life.",
    "I reserve my human right to hold my own beliefs and practice them as I see fit and allow other people the same right to believe and act as they wish.",
    "I prefer loving rewarding consensual relationships with beautiful positive and virtuous women.",
    "I believe men and women are different and that each has their own unique and important strengths and abilities.",
    "I believe men have the sacred duty to protect and provide for the important people in their lives.",
    "I believe men have the sacred duty to protect the innocence and sanctity of their children and reserve the right and responsibility to raise their children as they see best to ensure their long-term happiness and success.",
    "I believe that men have the sacred duty to raise strong capable and honorable sons.",
    "I believe that men have the sacred duty to raise kind feminine and virtuous daughters.",
    "I utterly disprove of violence within romantic or familiar relationships.",
    "I support good and honest governments who will obey their laws.",
    "I prefer to only conduct business dealings with trusted and vetted brothers.",
    "I believe that a man has a sacred duty to hold true to his word and do exactly what he says he will do.",
    "I believe that a man’s life is difficult and that he has a sacred duty to become strong to handle such difficulty.",
    "I believe men are personally responsible for their actions and for the results they achieve in their lives.",
    "I believe in emotional control and the vital need to become disciplined and professional in all things.",
    "I believe that men have the sacred duty to approach everything in life from a position strength.",
    "I believe that all men have the sacred duty to become men of upright and virtuous character and live above all possible reproach.",
    "I believe it is incumbent upon me to ruthlessly identify my own weaknesses and limitations and I eagerly work to overcome them and become more capable in all realms.",
    "I seek to improve my personal freedom to think, act and live to my masculine imperative in all ways.",
    "I believe each man has a sacred duty to mold the physical body into the strongest and most resilient and most capable version of itself.",
    "I believe that men have the sacred duty to rigorously train themselves both physically and mentally every day.",
    "I believe that I have the sacred duty to only eat the highest quality foods possible.",
    "I reserve the right to protect the sanctity of my bloodstream and make my own decisions about medical care and procedures.",
    "I affirm the importance of endlessly improving my mental faculties through diligent work, study and practice.",
    "I believe in acquiring wealth and abundance in order to improve my life and do good for those I care about.",
    "I believe in the merits of healthy competition and constantly encourage all men to seek out competition to improve themselves.",
    "I believe that masculine brotherhood is essential to men’s mental health, happiness and success. And I relentlessly encourage men to meet together, train together and work together.",
    "I maintain the trust of my brothers through reverent silence, regarding our most sacred and shared experiences.",
    "I believe in honoring my ancestors and living in a way that would make most of them proud of me today.",
    "I reserve the right to administer difficult rites of passage for our young men to allow them to earn the rank of manhood.",
    "I affirm the importance and need for travel and adventure as men.",
    "I seek to help men overcome poor mental health through embracing hard work, physical improvements and shared masculine brotherhood.",
    "I reserve the right to make the best choices I can at the time to protect myself and respect my mental health.",
    "I do good in the world and seek to help those less fortunate.",
    "I believe all men have the responsibility to lead and guide those they care about for greater health, prosperity, and happiness.",
    "I choose to only interact with those who are respectful and civil to me in return for my own respectfulness and civility.",
    "I reserve the freedom to speak and refer to others as I believe is best and most truthful.",
    "I reserve the right to choose my company and include only those whom I believe are best for my health, happiness and success.",
    "I believe I have an imperative to only spend my time that I determine is beneficial, uplifting and empowering to myself and others.",
    "I choose to only allow myself to be influenced by those who I believe have my best interests in mind.",
    "Each day I dedicate myself to create anew the greatest possible positive impact on the world and do the work necessary to achieve a greater masculine excellence across all realms of human endeavor."
    ]

Don't have twitter API so I used apify to scrape Andrew Tate quotes off of twitter. Here I'm putting everything in a nice dataframe.

In [ ]:
# create an empty list to store the data
rows = []

# iterate through the json data and extract the desired information
for item in tqdm(data):
    full_text = item["full_text"]
    if full_text[:2]=='RT' or full_text[:5]=='https':
        continue
    name = item["user"]["name"]
    date = item["created_at"]
    url = item["url"]
    # add the extracted data as a row to the list
    rows.append([name, full_text, date, url])

for tenet in tenets:
  rows.append(['Andrew Tate',tenet,'NA','NA'])

df = pd.DataFrame(rows, columns=["name", "full_text", "date", "url"])
df.to_csv(DIR+"tate.csv")

100%|██████████| 576/576 [00:00<00:00, 448472.08it/s]


In [ ]:
df[df["date"]!="NA"]

name                                          full_text  \
0    Andrew Tate  When Allah said ‘I test only those I love.’ I ...   
1    Andrew Tate  Most men are REACTIVE, they respond to problem...   
2    Andrew Tate  Do you deserve to be happy?\n\nWatch the full ...   
3    Andrew Tate  "Truth has come and falsehood has vanished. Fa...   
4    Andrew Tate  An Important Video Statement from our Lawyer.\...   
..           ...                                                ...   
277  Andrew Tate  It’s IMPOSSIBLE to not get everything you want...   
278  Andrew Tate  Happiness must be earned\n\nWatch the full int...   
279  Andrew Tate  Lol.\n\nEven in Romanian Prison they talk abou...   
280  Andrew Tate  You are the Matrix. You are an Agent.\n\nBut t...   
281  Andrew Tate  My whole life everyone called me paranoid. \n\...   

                         date  \
0    2023-01-10T20:56:12.000Z   
1    2023-01-10T15:52:54.000Z   
2    2023-01-10T14:40:15.000Z   
3    2023-01-09T20:55:04.000Z   
4    2023-01-09T13:52:52.000Z   
..                        ...   
277  2023-01-11T18:41:14.000Z   
278  2023-01-11T17:08:19.000Z   
279  2023-01-11T14:28:04.000Z   
280  2023-01-11T13:16:38.000Z   
281  2023-01-11T10:22:52.000Z   

                                                   url  
0    https://twitter.com/Cobratate/status/161291623...  
1    https://twitter.com/Cobratate/status/161283991...  
2    https://twitter.com/Cobratate/status/161282162...  
3    https://twitter.com/Cobratate/status/161255356...  
4    https://twitter.com/Cobratate/status/161244731...  
..                                                 ...  
277  https://twitter.com/Cobratate/status/161324466...  
278  https://twitter.com/Cobratate/status/161322127...  
279  https://twitter.com/Cobratate/status/161318095...  
280  https://twitter.com/Cobratate/status/161316297...  
281  https://twitter.com/Cobratate/status/161311924...  

[282 rows x 4 columns]

Goodreads is super easy to scrape. Managed to get a lot of Confucius quotes off of there.

In [ ]:
confucius = []
for i in tqdm(range(17)):
    url = "https://www.goodreads.com/author/quotes/15321.Confucius?page=" + f'{i+1}'
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
    quotes = soup.find_all('div',{'class':'quoteText'})
    for quote in quotes:
        text = re.sub(r'\n', '', str(quote)) # remove newline characters
        # text = re.sub(r'Confucious', '', text) # remove the word "Confucious"
        # text = re.sub(r'Confucius', '', text) # remove the word "Confucious"
        # text = re.sub(r'\s+', ' ', text) # remove extra spaces
        # text = re.sub(r'-', '',text)
        # text = re.sub(r'―', '',text)
        text = re.search(r'<div class="quoteText">(.+?)<br/>',text)
        text = str(text.group(1))
        text = text.strip() # remove whitespace from the beginning and end
        text = text.strip('“').strip('”')
        # text = re.findall(r'".+"', text)
        confucius.append(['Confucius',text,'NA',url])

df1 = pd.DataFrame(confucius, columns=["name", "full_text", "date", "url"])

100%|██████████| 17/17 [00:08<00:00,  2.01it/s]


Exported the scraped confucius quotes so I don't have to scrape everytime.

In [ ]:
# df1.to_csv(DIR + 'confucius.csv')

In [ ]:
df1.loc[12,'full_text']

'Respect yourself and others will respect you.'

# Similarity

Loading the sentence transformers model.

In [ ]:
# from sentence_transformers import SentenceTransformer
# model = SentenceTransformer('all-distilroberta-v1')

Downloading:   0%|          | 0.00/737 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/190 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/10.3k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/653 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/116 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/15.7k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/456k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/329M [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/53.0 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/239 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/333 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/13.1k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/798k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/349 [00:00<?, ?B/s]

In [ ]:
from scipy.spatial.distance import cosine

In [ ]:
df1.shape[0]*df.shape[0]

158593

In [ ]:
from IPython.display import clear_output

WARNING: The cell below will take a lot of time to run as there are 158,593 comparisons

Here I am comparing all of Confucius's quotes with all of Andrew Tate's quotes using the "all-distilroberta-v1" model.

In [ ]:
# similar_sentences = set()
r = []
count = 0
for i in tqdm(range(df.shape[0])):
  for j in range(df1.shape[0]):
    count +=1
    clear_output()
    print(count)
    embeddings = model.encode([df.loc[i,'full_text'],df1.loc[j,'full_text']])
    similarity = 1 - cosine(embeddings[0], embeddings[1])
    r.append([df.loc[i,'full_text'],df1.loc[j,'full_text'],similarity])
similar_sentences = pd.DataFrame(r, columns=["tate", "confucius","similarity"])

100%|██████████| 323/323 [42:47<00:00,  7.95s/it]

158593


Exporting the dataframe.

In [ ]:
# similar_sentences.to_csv(DIR+'similarity.csv')

In [12]:
similar_sentences = pd.read_csv(DIR+'similarity.csv')

Sorting by similarity.

In [13]:
sorted = similar_sentences.sort_values(by='similarity',ascending=False).reset_index()

In [15]:
sorted.head()

index  Unnamed: 0                                               tate  \
0  147251      147251  I believe it is incumbent upon me to ruthlessl...   
1   82505       82505  Reading books is for losers who are afraid to ...   
2  147255      147255  I believe it is incumbent upon me to ruthlessl...   
3   41430       41430  I don’t understand how any self-respecting man...   
4   82901       82901  Reading books is for losers who are afraid to ...   

                                           confucius  similarity  
0  It is true that we shall not be able to reach ...    0.663559  
1  You cannot open a book without learning someth...    0.615090  
2  The will to win, the desire to succeed, the ur...    0.611537  
3  A man should demand much from himself, but lit...    0.601160  
4  you cannot open a book without learning something    0.591984

In [14]:
q = {'Andrew Tate':set(),'Confucius':set()}
for i in range(900):
  q['Andrew Tate'].add(sorted.loc[i,'tate'])
  q['Confucius'].add(sorted.loc[i,'confucius'])

Some quotes addded manually.

In [ ]:
# manually added directly from tates account
# January 11, 2023 to January 27, 2023
quotes_manually_added_cobratate = [
    'A man without struggle is never going to be a powerful man. The best men you know are men that have been through struggle or depressed and have come out on the other side.If you’re going to be a hero, you’re going to suffer.',
    'Understand the energy of people can be STOLEN from them. If you make somebody angry, so that they think about you. While you refuse to think of them back. They are converting calories into thoughts that have your name on them. They are giving you power.',
    '''How many more "I'll get started tomorrow" days do you need? How much longer will you accept less than what you can have?''',
    'If the people in your circle don’t inspire you. They’re not your circle. They’re your cage.',
    'The Matrix will tell you that not everyone can become exceptional. My message is that with enough concentrated effort, anybody can.',
    'You don’t lack energy - you’re just an idiot.',
    'Every day is another chance to decide to change your life, yet you allow life to beat you down to your knees, distracting you from everything you want. Break the trance now.',
    'There is no light without dark.',
    '''Your future is the result of your daily actions, you're defined by what you do today. Lazy now, loser later. Get to work.''',
    "Lazy people do a little work and feel like they should be winning, winners work as hard as possible and still worry they're being lazy.",
    "All of life is a competition and the Rambos of the world will always lose to a team of men. Lone soldiers lose.",
    "Surely the winds are too strong, surely the cold bites. I'm here to tell you, keep climbing.",
    "Great men are meteors designed to burn so that earth may be lighted.",
    "I know talented men who are unsuccessful, I know smart men who are unsuccessful. I do not know a single man who's genuinely persistent that doesn't have a fantastic life. Persistence is the measure of man. It trumps talent and intelligence. If you never give up, you never lose.",
    "In the darkened silence of solitary confinement there is almost no noise. Pray for me, I can hear you.",
    "Life is easy for the professional.",
    "How many more \"I'll get started tomorrow\" days do you need? How much longer will you accept less than what you can have?",
    "Adversity builds men. It is your duty to challenge yourself and craft your own world.",
    "One man who stands by his word is worth more than 1,000 who don’t.",
    "Everything bad happens at once. That's how life works. Be prepared. Enjoy the storm. Sing in the rain. Look forward to the rainbow.",
    "You absorb and share the energy of the men around you, You completely underestimate how much your life is affected by the people you associate with. Energy transfer is real.",
    "A lot of people go through life with a very average attitude to life and they wait for their one big break... You have to build your big break. And you are going to do that with a thousand tiny victories.",
    "The world tries very hard to tell you NOT to be angry. Angry men get things done. Be angry you're a fat loser with no money. Then do something about it. Directional anger = superpower.",
    "You need to start believing you are the fucking man. Even if you're not the man yet. Even before I was World Champion I knew I was gonna be the man.",
    "You are underperforming. You know in your gut that you are. Will you ignore this signal from the universe? Or is today the day you become a professional? Decide.",
    "Solitary confinement provides the perfect environment to meditate and hone my dragon fist. Once they fail to kill me, I will emerge. Perpetrators beware.",
    "You must know where you’re going with your life. Money won’t save you. Only a vision for the future backed by balls to do whatever you must will.",
    "How I feel has no bearing on how I live my life.",
    "A man without a vision for his future always returns to his past",
    "Never seen somebody who didn’t believe in themselves pull it off. Literally never.",
    "Most of you never feel the joy of living with this level of permanent “we both die or nobody dies” backup. It’s what brothers are for.",
    "As soon as you are granted life, you are guaranteed death as well. What you do in between is up to you.",
    "You can want to be better all you want. “Wanting” means nothing. Only challenging yourself will break you free from the chains of your middle-brained thinking.",
    "Respect lies on the other side of fear. You must overcome this. Or else you’ll live your life as a coward for your entire human existence.",
    "Your life gets better only when you get better.",
    "You can cry and pray for more. You’ll be ignored. When you stand up and become worthy of everything you want? The world is yours.",
    "Men used to conquer foreign lands. Now we make money, train, build our bloodline, and more. Men must conquer something.",
    "The devil needs weak men. Honing power is homage to God.",
    "The true mark of bravery and honor is a king who can hire soldiers, but instead decides to fight his own battles.",
    "I will emerge from the dark roach-infested dungeon more ninja than ever before. They cannot break me",
    "Reality is self-constructed. In my world, I am in absolute control every second. Do you use this power?",
    "Seeds of doubt grow into forests of inaction.",
    "It’s called clown world because there will always be a large part of the population who believe in the circus.",
    "Pitch black surrounded by cockroaches. Unable to sleep, I hunt in the dark. I await daybreak to count my kills.",
    "I don't believe in faking it before you make it. There is nothing “fake” about belief. You have to wait for the rest of the world to catch up with what you already know. You have to see through time, act accordingly, and prepare a “told you so” that eternally upsets doubters.",
    "Knowing how to control yourself is never a disadvantage. It doesn't matter how savage the situation - becoming savage is a choice. Staying calm is a choice."
    ]

In [ ]:
# January 19, 2023 to January 24, 2023
quotes_manually_added_quotestopg = [
    "You are organized, fast and systemized. You stop doing things you know are a waste of time. You stop acting based on the emotional whims of nobodies. Imagine that version of yourself. Imagine what that man accomplishes in one year. You are exactly where you deserve to be.",
    "Poor? Then money isn't your problem, your brain is the problem. You are the reason for your own poverty.",
    "You get depressed because you aren't where you want to be in life. Maybe you're broke. Maybe you're out of shape. Maybe you just want a different life altogether.",
    "While others complain they're struggling because they don't feel happy enough, I'm simply happy to be struggling. Overcoming impossible odds is the beauty of line as a man. I don't want to be happy. I want to be great.",
    "When you believe things that are blatantly false, it means you're open to believing anything. The perfect slave is one who does not listen to his own eyes. Wake up!",
    "When is the last time you've ignored fear and acted anyway? Only bravery leads to riches. Cowardice is disgusting and will always be punished by God.",
    "Understand that as people become poor, the money hasn't vanished. Someone else is becoming rich. Yin and yang, the balance of the universe.",
    "Think of all the amazing things you could've accomplished by now if you weren't so brutally lazy. Imagine all the time you wasted, every opportunity you let pass by, all the good advice you arrogantly ignored.",
    "They call it heart break, even though your heart works fine. If they called it mind break, you'd understand it can be controlled. Along with everything else.",
    "The top of one mountain is the bottom of the next.",
    "The quitter is the most useless class of man. Completely useless. He could be the most talented man. God could have blessed this man with absolutely every single talent and gift possible.",
    "Sloppy in one area of life results in sloppiness in all areas of your life. Sloppiness is a mindset. When you live your life in constant mess and disorganization, how can you expect to be a professional at anything else? How are you going to conquer earth?",
    "Saving cash to lose time is a subliminal message that your time isn't effective. You already waste time, so you don't need to pay for more.",
    "Peasants have no concern for endless monumental achievement because they know they are peasants. When you know you are a god, you expect endless godly achievement from yourself.",
    "No exceptional person ever lived like an average person.",
    "My dad always used to say 'outstanding in the rain' as opposed to simply 'outstanding' when he was impressed. I asked him: Dad, why the rain? And he replied: Son, any jackass can perform when the sun is shining.",
    "Most successful people didn't enjoy doing what made them successful. However, they enjoyed being good at whatever they were doing. Why do something and not try? Why do something and not be the best? Winners understand this. Losers do not.",
    "Lacking motivation is not a real problem. Only absolute losers 'suffer' with the inability to strive to become something.",
    "Intellect is nothing without energy. Ideas are nothing without energy. High energy people win.",
    "I find that very few men really want freedom. They'd rather play it safe. And take the highest hourly wage they can get their hands on. Hoping that will make them rich when the math clearly doesn't add up.",
    "Happiness is a lack of grief. If you're not too distraught to function, you're happy. Simple. Stop aiming for the kiddish childish joy of a Christmas morning. We're adults. Just be satisfied nobody you love has died today. That's happy enough.",
    "Every action you take is molding who you are as a person. Every time you sleep in, every time you are undisciplined, you are training yourself that it's okay. A downward spiral to mediocrity.",
    "It doesn't matter who you are, it doesn't matter what you say, there are some people who don't like you. And the people who don't like you are not interested in the truth.", "Imagine looking in the mirror and being proud of yourself.",
    "Even from a very young age i was taught that going to authority and crying and hoping someone else was gonna fix your problems isn't the way a man should conduct himself.", "As a man trauma and difficulty are extremely important because they're the building blocks for mental fortitude and physical fortitude.",
    "There's no one on this planet who can do shit i couldn't do if i put my mind to it.",
    "Stick to your word even if there is zero foreseeable benefit for doing so. It's always worth it, in the end.",
    "Even before i was a world champion, i knew i was gonna be the man.",
    "Your mind must be stronger than your feelings.",
    "Your deeper meaning already exists as a man. Become bigger, stronger, faster, richer, and more powerful. That's your meaning. And you already know that. And you're hiding from it and looking for another path because you're a coward and a weakling. Go Train!",
    "You're on your own. Your politicians cannot save you from yourself.",
    "You are on your own. The sooner you understand only you can improve your life, the better.",
    "You're not thankful for something if you don't fight to keep it.",
    "You control the beliefs you hold. Why choose to hold the ones which affect your life negatively?",
    "You are exactly where you deserve to be. Change who you are and you will change how you live.",
    "Who do you speak to on a daily basis? Who are you learning from? Who is molding your mind? All the men that came to mind, combine them and understand that is where you're headed as a man. That is the man you are slowly becoming.",
    "There's a universe where from today onwards you became the epitome of professionalism. Every step is purposeful. Every move is quickly and efficiently calculated.", "Revenge is sweet because it's a battle. Battle is ultra addictive.",
    "Most people have no real control of their life and are entirely uninterested in the idea of autonomy.",
    "If the punishment for a 'crime' is a fine. Then it's an activity which is legal for a price."
]

Adding the manually added quotes.

In [ ]:
for i in range(len(quotes_manually_added_cobratate)):
  q['Andrew Tate'].add(quotes_manually_added_cobratate[i])

In [ ]:
for i in range(len(quotes_manually_added_quotestopg)):
  q['Andrew Tate'].add(quotes_manually_added_quotestopg[i])

In [ ]:
q['Andrew Tate'] = list(q['Andrew Tate'])
q['Confucius'] = list(q['Confucius'])

In [ ]:
json_string = json.dumps(q)
with open(DIR+'quotes.json', 'w') as f:
    f.write(json_string)

Final Quotes

In [ ]:
json_string = json.dumps({"Andrew Tate":["When's the last time you've sat and discussed how to become rich? If you don’t think about it. And you don’t speak about it. How is it ever going to happen? Speak it into existence.","A lot of people go through life with a very average attitude to life and they wait for their one big break... You have to build your big break. And you are going to do that with a thousand tiny victories.","Everything you do creates a change in the Universe. Yet most of you have been programmed to use this truth against your best interest.","I don't believe in faking it before you make it. There is nothing “fake” about belief. You have to wait for the rest of the world to catch up with what you already know. You have to see through time, act accordingly, and prepare a “told you so” that eternally upsets doubters.","Something has shifted in the Universe. God has blessed many of you with the chance you'd been hoping for. Will you notice it before it's too late? The Universe can not give its graces to you all. Tick Tok.","When you believe things that are blatantly false, it means you're open to believing anything. The perfect slave is one who does not listen to his own eyes. Wake up!","Even before I was a world champion, I knew I was gonna be the man.","Your life path was unique. But you are not unique. You have ignored the unique and beautiful lessons god has given you via trauma. You refused to analyse, absorb and WORK post lessons. So despite your unique life experience, you’re still very average. You’re a loser.","Life is easy for the professional.","Hard times are here. There's no more easy money out there to take. All the amateurs will be exposed. This is the time to become a professional. You only have two choices.","I seek to improve my personal freedom to think, act and live to my masculine imperative in all ways.","A man can be in the worst position and all he will say is “I’ll take care of it”. He might be flat broke with no hope, but that’s what he will say. And might get grumpy but somehow he always finds a way. Respect to all the real men out there that do the impossible daily.","I believe it is incumbent upon me to ruthlessly identify my own weaknesses and limitations and I eagerly work to overcome them and become more capable in all realms.","Some men are born with a fire inside of them.","There is no light without dark.","Be yourself. Everyone else is taken.","I believe men have the sacred duty to protect and provide for the important people in their lives.","You absorb and share the energy of the men around you, You completely underestimate how much your life is affected by the people you associate with. Energy transfer is real.","I reserve the right to administer difficult rites of passage for our young men to allow them to earn the rank of manhood.","I believe men are personally responsible for their actions and for the results they achieve in their lives.","You get depressed because you aren't where you want to be in life. Maybe you're broke. Maybe you're out of shape. Maybe you just want a different life altogether.","Your life is what it’s like because of the influences you have. You can’t not be influenced, but you can control what influences you. Do you understand?","It’s called clown world because there will always be a large part of the population who believe in the circus.","Every action you take is molding who you are as a person. Every time you sleep in, every time you are undisciplined, you are training yourself that it's okay. A downward spiral to mediocrity.","I believe that all men have the sacred duty to become men of upright and virtuous character and live above all possible reproach.","My dad always used to say 'outstanding in the rain' as opposed to simply 'outstanding' when he was impressed. I asked him: Dad, why the rain? And he replied: Son, any jackass can perform when the sun is shining.","I don't believe in faking it before you make it. There is nothing “fake” about belief. You have to wait for the rest of the world to catch up with what you already know. You have to see through time, act accordingly, and prepare a “told you so” that eternally upsets doubters.","If the people in your circle don’t inspire you. They’re not your circle. They’re your cage.","I believe that men have the divine imperative to become as capable, powerful and competent as possible in this life.","Knowing how to control yourself is never a disadvantage. It doesn't matter how savage the situation - becoming savage is a choice. Staying calm is a choice.","Mental resilience is trained. It is practiced. Like all else of virtue. You must fortify your mind against negative thoughts. I can’t even remember how it feels to doubt myself.","No exceptional person ever lived like an average person.","I believe that a man has a sacred duty to hold true to his word and do exactly what he says he will do.","Imagine looking in the mirror and being proud of yourself.","Revenge is sweet because it's a battle. Battle is ultra addictive.","I know talented men who are unsuccessful, I know smart men who are unsuccessful. I do not know a single man who's genuinely persistent that doesn't have a fantastic life. Persistence is the measure of man. It trumps talent and intelligence. If you never give up, you never lose.","You are organized, fast and systemized. You stop doing things you know are a waste of time. You stop acting based on the emotional whims of nobodies. Imagine that version of yourself. Imagine what that man accomplishes in one year. You are exactly where you deserve to be.","You’ve been programmed to be poor and weak. Easily controlled. However, you can also be programmed to be rich and strong.","I believe in acquiring wealth and abundance in order to improve my life and do good for those I care about.","I believe that a man’s life is difficult and that he has a sacred duty to become strong to handle such difficulty.","You control the beliefs you hold. Why choose to hold the ones which affect your life negatively?","Intellect is nothing without energy. Ideas are nothing without energy. High energy people win.","You need to start believing you are the fucking man. Even if you're not the man yet. Even before I was World Champion I knew I was gonna be the man.","Ghengis Khan had endless woman and 200 children as reward for conquest.\n\nI am the most searched man on the planet.\n\nI have conquered Earth.\n\nI am the highest status male on the planet.\n\nFemales do not expect loyalty from me.\n\nThey only expect that of lesser men.","Real work is absolutely admirable. \n\nHard-working men make the world go round.","Men go through so much pain that they will never talk about it because they know that nobody cares.","I believe each man has a sacred duty to mold the physical body into the strongest and most resilient and most capable version of itself.","As a man trauma and difficulty are extremely important because they're the building blocks for mental fortitude and physical fortitude.","HARAM.","Saving cash to lose time is a subliminal message that your time isn't effective. You already waste time, so you don't need to pay for more.","I believe that I have the sacred duty to only eat the highest quality foods possible.","You get songs stuck in your head. \n\nI get ideas stuck in my head. \n\nDo you understand?","I affirm the importance and need for travel and adventure as men.","You can want to be better all you want. “Wanting” means nothing. Only challenging yourself will break you free from the chains of your middle-brained thinking.","MashAllah","You can fight the realities of the world and lose. \n\nOr you can accept them and become successful.\n\nYou decide.","Solitary confinement provides the perfect environment to meditate and hone my dragon fist. Once they fail to kill me, I will emerge. Perpetrators beware.","You are on your own. The sooner you understand only you can improve your life, the better.","I reserve the right to make the best choices I can at the time to protect myself and respect my mental health.","I reserve the freedom to speak and refer to others as I believe is best and most truthful.","My team is united against the demons.","I do good in the world and seek to help those less fortunate.","Never seen somebody who didn’t believe in themselves pull it off. Literally never.","I believe men have the sacred duty to protect the innocence and sanctity of their children and reserve the right and responsibility to raise their children as they see best to ensure their long-term happiness and success.","I have truly conquered the world.","We are in the midst of a spiritual war. \n\nThe final battle between humanity and the machines - before you’re reduced to a QR code. \n\n99.9% of men refuse the sword and shield, paralyzed as civilians, awaiting slaughter from the machine horde. \n\nThe real pandemic is cowardice.","I maintain the trust of my brothers through reverent silence, regarding our most sacred and shared experiences.","In the darkened silence of solitary confinement there is almost no noise. Pray for me, I can hear you.","The answer is out there. It's looking for you. And it will find you if you want it to.","The world tries very hard to tell you NOT to be angry. Angry men get things done. Be angry you're a fat loser with no money. Then do something about it. Directional anger = superpower.","I will emerge from the dark roach-infested dungeon more ninja than ever before. They cannot break me","It doesn't matter who you are, it doesn't matter what you say, there are some people who don't like you. And the people who don't like you are not interested in the truth.","You are exactly where you deserve to be. Change who you are and you will change how you live.","Cowardice is the modern plague of man.","You're on your own. Your politicians cannot save you from yourself.","You can cry and pray for more. You’ll be ignored. When you stand up and become worthy of everything you want? The world is yours.","Respect lies on the other side of fear. You must overcome this. Or else you’ll live your life as a coward for your entire human existence.","As soon as you are granted life, you are guaranteed death as well. What you do in between is up to you.","You get the life you deserve.\n\nMost of you are cowards and so that is what you get.\n\nIf you want more?\n\nDo more and become more.","Create a life you’re proud of.\n\nCrafting every aspect of your reality with intention.\n\nAre you awake? Do you KNOW what you are doing?\n\nYes or no.","Most of you never feel the joy of living with this level of permanent “we both die or nobody dies” backup. It’s what brothers are for.","A man without a vision for his future always returns to his past","Each day I dedicate myself to create anew the greatest possible positive impact on the world and do the work necessary to achieve a greater masculine excellence across all realms of human endeavor.","Sloppy in one area of life results in sloppiness in all areas of your life. Sloppiness is a mindset. When you live your life in constant mess and disorganization, how can you expect to be a professional at anything else? How are you going to conquer earth?","Your deeper meaning already exists as a man. Become bigger, stronger, faster, richer, and more powerful. That's your meaning. And you already know that. And you're hiding from it and looking for another path because you're a coward and a weakling. Go Train!","I believe that men have the sacred duty to approach everything in life from a position strength.","Even from a very young age i was taught that going to authority and crying and hoping someone else was gonna fix your problems isn't the way a man should conduct himself.","Know this. Not everybody will rise today. Some will die as they sleep. Most will rise and their day will be wasted. Very few will rise and shine. I do both. Day after fucking day. I never waste a conscious second.","Great men are meteors designed to burn so that earth may be lighted.","Think of all the amazing things you could've accomplished by now if you weren't so brutally lazy. Imagine all the time you wasted, every opportunity you let pass by, all the good advice you arrogantly ignored.","You said tomorrow, yesterday.","I reserve my human right to hold my own beliefs and practice them as I see fit and allow other people the same right to believe and act as they wish.","Peasants have no concern for endless monumental achievement because they know they are peasants. When you know you are a god, you expect endless godly achievement from yourself.","How many more \"I'll get started tomorrow\" days do you need? How much longer will you accept less than what you can have?","Most men are reactive, they respond to problems as they happen. \n\nYou must be preemptive at the highest echelons of power.","All of life is a competition and the Rambos of the world will always lose to a team of men. Lone soldiers lose.","While others complain they're struggling because they don't feel happy enough, I'm simply happy to be struggling. Overcoming impossible odds is the beauty of line as a man. I don't want to be happy. I want to be great.","The quitter is the most useless class of man. Completely useless. He could be the most talented man. God could have blessed this man with absolutely every single talent and gift possible.","I believe in emotional control and the vital need to become disciplined and professional in all things.","Everything bad happens at once. That's how life works. Be prepared. Enjoy the storm. Sing in the rain. Look forward to the rainbow.","There is a small percentage of men in the world who want more.\n\nMen who refuse to live a normal life, who refuse to be average or just above average.\n\nMen who refuse to be anything less than spectacular.\n\nThere is a man out there who lives the life of YOUR dreams.","Your life gets better only when you get better.","Death smiles at us all. \n\nAll a man can do is smile back.","I believe all men have the responsibility to lead and guide those they care about for greater health, prosperity, and happiness.","Most men sleep fine knowing their dreams will remain dreams.\n\nBut some men, refuse to have anything less than everything\n\nWe will either live the life of our dreams or die trying.\n\nWill you?","It’s impossible to not get everything you want if you truly work hard every day. \n\nMost of you just don’t work hard. \n\nThe few who do, work hard on the wrong things. \n\nFix this.","Knowing how to control yourself is never a disadvantage. It doesn't matter how savage the situation - becoming savage is a choice. Staying calm is a choice.","Your future is the result of your daily actions, you're defined by what you do today. Lazy now, loser later. Get to work.","I believe in the merits of healthy competition and constantly encourage all men to seek out competition to improve themselves.","Most successful people didn't enjoy doing what made them successful. However, they enjoyed being good at whatever they were doing. Why do something and not try? Why do something and not be the best? Winners understand this. Losers do not.","Surely the winds are too strong, surely the cold bites. I'm here to tell you, keep climbing.","Most people overestimate the time it takes to accomplish things.\n\nYou can accomplish what most people take years to do within months.\n\nIt's something you need to feel to believe.","You have no option but to perform at the highest levels. \n\nAnything less is failure.","Money is an amplifier.\n\nSee a rich dork get out of a Lambo. Super dork.\nSee a huge mafia guy get out of a Lambo. Killer. \n\nThose who say money doesnt make you happy - were always miserable.\n\nThose who were happy broke - are very happy rich.","Adversity builds men. It is your duty to challenge yourself and craft your own world.","They call it heart break, even though your heart works fine. If they called it mind break, you'd understand it can be controlled. Along with everything else.","I don’t understand how any self-respecting man can expect anything less than the absolute best of everything","A real brother always has your back.","Men used to conquer foreign lands. Now we make money, train, build our bloodline, and more. Men must conquer something.","I believe that men have the sacred duty to rigorously train themselves both physically and mentally every day.","The Matrix will only tell you what the Matrix needs you to believe.","Take time to actually analyze why you do the things you do. Look at every decision and every belief about the world you hold. Now, look at everyone who influenced those. Are they where you want to be? Is that who you want to be like?","There are dark forces working against you. \n\nYou must recognize them and you must fight back.","I support good and honest governments who will obey their laws.","A man without struggle is never going to be a powerful man. The best men you know are men that have been through struggle or depressed and have come out on the other side.If you’re going to be a hero, you’re going to suffer.","I find that very few men really want freedom. They'd rather play it safe. And take the highest hourly wage they can get their hands on. Hoping that will make them rich when the math clearly doesn't add up.","I know of no better life purpose than to perish in attempting the great and impossible.","There's no one on this planet who can do shit i couldn't do if i put my mind to it.","I believe in honoring my ancestors and living in a way that would make most of them proud of me today.","Poor? Then money isn't your problem, your brain is the problem. You are the reason for your own poverty.","You know something. What you know you can't explain, but you feel it. You've felt it your entire life. You know there's something wrong with the world, it’s a a splinter in your mind, driving you mad. It is this feeling that has brought you to me.","Stick to your word even if there is zero foreseeable benefit for doing so. It's always worth it, in the end.","Understand the energy of people can be STOLEN from them. If you make somebody angry, so that they think about you. While you refuse to think of them back. They are converting calories into thoughts that have your name on them. They are giving you power.","I believe that men have the sacred duty to raise kind feminine and virtuous daughters.","Men are expected to have money, intelligence, and other achievements. Anything less? You’re called a loser. Understand, life isn’t fair. But that’s a good thing.","I believe I have an imperative to only spend my time that I determine is beneficial, uplifting and empowering to myself and others.","Every period of history had men who conquered the known world.\n\nMen with undeniable superiority over their peers.\n\nNames spoken across the globe.\n\nThese men owned culture.\n\nGenghis Khan\nAtilla The Hun\n\nYou are witnessing the modern day equivalent in his prime.","When is the last time you've ignored fear and acted anyway? Only bravery leads to riches. Cowardice is disgusting and will always be punished by God.","You're not thankful for something if you don't fight to keep it.","I will never forgive myself for failure. \n\nI am absolutely hard on myself. \n\nI never ever forgive myself. \n\nFailing is permanent pain. \n\nFor the rest of my human years. \n\nThis is why I never fail.","The rules are designed to prevent you from freedom. \n\nThey’re so concisely derived that even if you break them - you stand near zero chance. \n\nYou must change the game that you’re playing entirely.","You are underperforming. You know in your gut that you are. Will you ignore this signal from the universe? Or is today the day you become a professional? Decide.","Who do you speak to on a daily basis? Who are you learning from? Who is molding your mind? All the men that came to mind, combine them and understand that is where you're headed as a man. That is the man you are slowly becoming.","Seeds of doubt grow into forests of inaction.","The phrase “be like water” doesn’t only mean adapting to any situation. \n\nIt means to be immensely powerful when directed, \n\nextremely cost-intensive for your enemies to contain, \n\nto be beautiful when calm \n\nand feared when angry. \n\nTo be absolutely necessary for life.","You are being programmed.\n\nYou cannot stop it.\n\nEverything you ARE stems from what you’ve experienced.\n\nIt’s an unstoppable force like Gravity.\n\nYou cannot stop the programming but,\n\nYou can choose what you hear and experience.\n\nYou can choose what and who programs your mind.","Lacking motivation is not a real problem. Only absolute losers 'suffer' with the inability to strive to become something.","Your mind must be stronger than your feelings.","If the punishment for a 'crime' is a fine. Then it's an activity which is legal for a price.","Make your last name mean something. Only a coward doesn’t care.","I believe that masculine brotherhood is essential to men’s mental health, happiness and success. And I relentlessly encourage men to meet together, train together and work together.","Pitch black surrounded by cockroaches. Unable to sleep, I hunt in the dark. I await daybreak to count my kills.","Happiness is a lack of grief. If you're not too distraught to function, you're happy. Simple. Stop aiming for the kiddish childish joy of a Christmas morning. We're adults. Just be satisfied nobody you love has died today. That's happy enough.","You must know where you’re going with your life. Money won’t save you. Only a vision for the future backed by balls to do whatever you must will.","One man who stands by his word is worth more than 1,000 who don’t.","Understand that as people become poor, the money hasn't vanished. Someone else is becoming rich. Yin and yang, the balance of the universe.","Reading books is for losers who are afraid to learn from life. \n\nSo they try and learn from the life others have lived. \n\nBut you never really learn unless you lived it. \n\nYou must feel it to believe it. \n\nBooks are a total waste of time.\n\nEducation for cowards.","I thank God my life was shit. \n\nIt makes having a great life so much better. \n\nThere is no light without dark.\n\nI live the best life a young strong man can possibly live.","Be savage in your approach towards what you want in life. No more cowardly attempts. Fully on, not half on.","If you treat your body like shit, you don't respect yourself. If you have friends who take advantage of you, you don't respect yourself. If you waste your own time playing video games, you don't respect yourself. If you don't want to be rich. You don't respect yourself.","There is no honor in comparing yourself to losers. You don’t want to be a winner amongst losers. You want to be a winner amongst winners.","Every day is another chance to decide to change your life, yet you allow life to beat you down to your knees, distracting you from everything you want. Break the trance now.","This is what people dont understand.\n\nBroke people dont have anything.\n\nThose in the middle try and have as much stuff as possible.\n\nAnd the truly wealthy dont \"have\" anything.","You don’t lack energy - you’re just an idiot.","I seek to help men overcome poor mental health through embracing hard work, physical improvements and shared masculine brotherhood.","I believe men and women are different and that each has their own unique and important strengths and abilities.","I prefer loving rewarding consensual relationships with beautiful positive and virtuous women.","If you’re truly obsessed with working on yourself. \n\nYou have very little time to find flaws in others.","The devil needs weak men. Honing power is homage to God.","Competence satisfies your soul as a man.","The Matrix will tell you that not everyone can become exceptional. My message is that with enough concentrated effort, anybody can.","The true mark of bravery and honor is a king who can hire soldiers, but instead decides to fight his own battles.","I choose to only interact with those who are respectful and civil to me in return for my own respectfulness and civility.","It appears the insanity of the ruling elite is exposed worldwide now. They have one command. Do not speak out, behave as sheep.","Know very well when to be extremely cautious and when to be extremely brave. \n\nThe problem with the modern “man” is he applies these two mindsets absolutely incorrectly. \n\nThey apply the wrong mindset every single time. \n\nSociety has programmed this way.","Reality is self-constructed. In my world, I am in absolute control every second. Do you use this power?","There's a universe where from today onwards you became the epitome of professionalism. Every step is purposeful. Every move is quickly and efficiently calculated.","I was made for battle. A warrior of the light. My enemies hope to attack me with lies. They try to destroy the family unit, they try to silence any loud opposition. I refuse to leave fellow men in the dark. Someone needs to show them the light.","Lone wolves survive. Packs thrive. You have no connections, you have no powerful or skillful friends. How do you expect to know what's happening all around you when you can only look one direction at a time?","I choose to only allow myself to be influenced by those who I believe have my best interests in mind.","Lazy people do a little work and feel like they should be winning, winners work as hard as possible and still worry they're being lazy.","I affirm the importance of endlessly improving my mental faculties through diligent work, study and practice.","You are poor. You are unimportant. Men do not fear you. Your woman disagrees with you. Your lives are shit. If I was forced to endure a year of your life it would be the worst level of depression imaginable.","The top of one mountain is the bottom of the next.","I believe that men have the sacred duty to raise strong capable and honorable sons.","Females have an evolutionary distain for weakness. Experiencing masculine frailty fills them with rage. The weak man compensates by finding a focus for the females aggression, to avoid suffering it himself.","Most people have no real control of their life and are entirely uninterested in the idea of autonomy.","How I feel has no bearing on how I live my life.","Your mind is an instrument. \n\nLearn to be its master, not its slave.","It seems every generation's great revolutionaries suffer from unfair imprisonment."],"Confucius":["No matter how busy you think you are, you must find time for reading, or surrender yourself to self-chosen ignorance.","The way of the superior person is threefold; virtuous, they are free from anxieties; wise they are free from perplexities; and bold they are free from fear.","To be able under all circumstances to practice five things constitutes perfect virtue; these five things are gravity, generosity of soul, sincerity, earnestness and kindness.","It is when those who are not strong enough have made some moderate amount of progress that they fail and give up...","You cannot open a book without learning something","The Master said, “To study, and then in a timely fashion to practice what you have learned—is this not satisfying? To have companions arrive from afar—is this not a joy? To remain unrecognized by others and yet remain free of resentment—is this not the mark of the gentleman?”<br>(Analects 1.1)","\"When good government prevails in the empire, ceremonies, music, and punitive military expeditions proceed from the son of Heaven. When bad government prevails in the empire, ceremonies, music, and punitive military expeditions proceed from the princes. When these things proceed from the princes, as a rule, the cases will be few in which they do not lose their power in ten generations. When they proceed from the great officers of the princes, as a rule, the case will be few in which they do not lose their power in five generations. When the subsidiary ministers of the great officers hold in their grasp the orders of the state, as a rule the cases will be few in which they do not lose their power in three generations. When right principles prevail in the kingdom, government will not be in the hands of the great officers. When right principles prevail in the kingdom, there will be no discussions among the common people.","Not to talk with people although they can be talked with is to waste people. To talk with people although they can't be talked with is to waste words. A man of understanding does not waste people, but he also does not waste words.","The superior man acts before he speaks, and afterwards speaks according to his actions.","Learn as if you were not reaching your goal and as though you were scared of missing it","They must often change who would remain constant in happiness and wisdom.","The superior man understands what is right; the inferior man understands what will sell.","It is true that we shall not be able to reach perfection, but in our struggle toward it we shall strengthen our characters and give stability to our ideas, so that, whilst ever advancing calmly in the same direction, we shall be rendered capable of applying the faculties with which we have been gifted to the best possible account.","And remember, no matter where you go, there you are.","Humankind differs from the animals only by a little and most people throw that away.","To know what is first and what is last will lead near to what is taught in the Great Learning.","He who studies but does not think is lost; he who thinks but does not study is dangerous.","When you know a thing, to hold that you know it; and when you do not know a thing, to allow that you do not know it--this is knowledge.","There are few men under heaven who can love and see the defects, or hate and see the excellence of an object.","Music produces a kind of pleasure which human nature cannot do without.","A man should say, I am not concerned that I have no place, I am concerned how I may fit myself for one. I am not concerned that I am not known, I seek to be worthy to be known.","I want you to be everything that's you, deep at the center of your being.”<br>“Life is really simple, but we insist on making it complicated","Those who are true in my village conduct themselves differently. A father covers for his son, and a son covers for his father. And being true lies in this.","When you have faults, do not fear to abandon them.","When you see a good person, think of becoming like her/him. When you see someone not so good, reflect on your own weak points.","The nobler sort of man emphasizes the good qualities in others, and does not accentuate the bad. The inferior does.","The man of wisdom is never of two minds;<br>the man of benevolence never worries;<br>the man of courage is never afraid.","You are not yet able to serve men, how can you serve spirits?","Let a ruler base his government upon virtuous principles, and he will be like the pole-star, which remains steadfast in its place, while all the host of stars turn towards it","Learning without thought is labour lost; thought without learning is perilous.","A true gentleman is one who has set his heart upon the Way. A fellow who is ashamed merely of shabby clothing or modest meals is not even worth conversing with.","I daily examine myself on three points:— whether, in transacting business for others, I may have been not faithful;— whether, in intercourse with friends, I may have been not sincere;— whether I may have not mastered and practised the instructions of my teacher.","What you do not want done to yourself, do not do to others.","The book salesman should be honored because he brings to our attention, as a rule, the very books we need most and neglect most.","For this reason the gentleman will employ a man on a distant mission and observe his degree of loyalty, will employ him close at hand and observe his degree of respect. He will hand him troublesome affairs and observe how well he manages them, will suddenly ask his advice and observe how wisely he answers. He will exact some difficult promise from him and see how well he keeps it, turn over funds to him and see with what benevolence he dispenses them, inform him of the danger he is in and note how faithful he is to his duties. He will get him drunk with wine and observe how well he handles himself, place him in mixed company and see what effect beauty has upon him. By applying these nine tests, you may determine who is the unworthy man.","The gentleman is dignified but not arrogant. The small man is arrogant but not digified","If your desire is for good, the people will be good.","No matter how busy you make think you are you must find time for reading, or surrender yourself to self-chosen ignorance","You do not understand even life. How can you understand death?","A man of humanity is one who, in seeking to establish himself, finds a foothold for others and who, in desiring attaining himself, helps others to attain.","You yourself desire rank and standing; then help others to get rank and standing. You want to turn your merits to account; then help others to turn theirs to account.","By nature, men are nearly alike; by practice, they get to be wide apart.","How dare I claim to be a sage or a benevolent man?","The noble-minded encourage what is beautiful in people and discourage what is ugly in them. Little people do just the opposite.","Real knowledge is to know the extent of one's ignorance.","If either wealth or poverty are come by honesty, there is no shame.","When anger rises, think of the consequences.","Nothing is more visible than things hidden; Nothing is more manifest than things minute; Therefore, the superior man should be aware of his aloneness","Do not be desirous of having things done quickly. Do not look at small advantages. Desire to have things done quickly prevents their being done thoroughly. Looking at small advantages prevents great affairs from being acccomplished.","The ultimate revenge is living well and being happy. Hateful people can't stand happy people. Before you embark on a journey of revenge, dig two graves.”-","That a man lives is because he is straight. That a man who dupes others survives is because he has been fortunate enough to be spared.","The expectations of life depend upon diligence the mechanic that would perfect his work must first sharpen his tools. ","Education breeds confidence. Confidence breeds hope. Hope breeds peace.","He who learns but does not think, is lost! He who thinks but does not learn is in great danger","He who conquers himself is the mightiest warrior.","The superior man, in the world, does not set his mind either for anything, or against anything; what is right he will follow.","There are those who act without knowing; I will have none of this. To hear a lot, choose the good, and follow it, to see a lot and learn to recognize it: this is next knowledge.","Life is really simple, but we insist on making it complicated.","To rank the effort above the prize may be called love.","Be not ashamed of mistakes and thus make them crimes.","When we see men of worth, we should think of equalling them; when we see men of a contrary character, we should turn inwards and examine ourselves.","A man should demand much from himself, but little from others. When you meet a man of worth, think how you may attain to his excellence. When you meet an unworthy one, then look within and examine yourself.","When anger arises, think of the consequences","Poor food and water for dinner, a bent arm for a pillow – that is where joy resides. For me, wealth and renown without honor are nothing but drifting clouds.","Imagination is more important than knowledge.","The Master said, “If your conduct is determined solely by considerations of profit you will arouse great resentment.","Learning without thinking is wasted effort; thinking without learning is dangerous","A man whose strength gives out collapses along the course. In your case you set the limits beforehand.","As the water shapes itself to the vessel that contains it, so a wise man adapts himself to circumstances.","The superior man is distressed by the limitations of his ability; he is not distressed by the fact that men do not recognize the ability that he has.","If you don't want to do something, don't impose on others","We take greater pains to persuade others that we are happy than in endeavoring to think so ourselves","Not to discuss with a man worthy of conversation is to waste the man. To discuss with a man not worthy of conversation is to waste words. The wise waste neither men nor words.","When respect is shown according to what is proper, one keeps far from shame and disgrace.","Faced with what is right, to leave it undone shows a lack of courage.","roads were made for journeys not destinations","I want you to be everything that's you, deep at the center of your being.","The nobler type of man is broad-minded and not prejudiced. The inferior man is prejudiced and not broad-minded.","To study and constantly, is this not a pleasure? To have friends come from far away places, is this not a joy? If people do not recognize your worth, but this does not worry you, are you not a true gentleman?","If there were an honorable way to get rich, I’d do it, even if it meant being a stooge standing around with a whip. But there isn’t an honorable way, so I just do what I like.","Everything has beauty, but not everyone sees it.","Only the wisest and stupidest of men never change.","The cautious seldom err.","Do not worry people not knowing you but strive so that you may worth knowing","When we see men of a contrary character, we should turn inwards and examine ourselves.","Without feelings of respect, what is there to distinguish men from beasts?","Don’t worry if you have no position: worry about making yourself worthy of one. Don’t worry if you aren’t known and admired: devote yourself to a life that deserves admiration.","Every man has two lives, and the second starts when he realizes he has just one","Wishing to regulate their families, they first cultivated their persons. Wishing to cultivate their persons, they first rectified their hearts. Wishing to rectify their hearts, they first sought to be sincere in their thoughts. Wishing to be sincere in their thoughts, they first extended to the utmost their knowledge.","Act with kindness but do not expect gratitude.","Choose a job you love, and you will never have to work a day in your life","The wise never doubt. The Humane never worry. The brave never fear.","The superior man thinks always of virtue; the common man thinks of comfort.","If you expect great things of yourself and demand little of others, you’ll keep resentment far away.","I used to take on trust a man's deeds after having listened to his words. Now having listened to a man's words I go on to observe his deeds.","Respect the gods and the devils but keep them at a distance","Wishing to regulate their families, they first cultivated their persons. Wishing to cultivate their persons, they first rectified their hearts. Wishing to rectify their hearts, they first sought to be sincere in their thoughts. Wishing to be sincere in their thoughts, they first extended to the utmost their knowledge. Such extension of knowledge lay in the investigation of things.","The superior man bends his attention to what is radical. That being established, all practical courses naturally grow up.","Wisdom, compassion, and courage are the three universally recognized moral qualities of man.","The superior man has a dignified ease without pride. The mean man has pride without a dignified ease.","The superior man, when resting in safety, does not forget that danger may come. When in a state of security he does not forget the possibility of ruin. When all is orderly, he does not forget that disorder may come. Thus his person is not endangered, and his States and all their clans are preserved.","What the superior man seeks is in himself; what the small man seeks is in others.","What you don´t want done to yourself, do not do to others.","Learn as though you would never be able to master it.","Walking among three people, I find my teacher among them. I choose that which is good in them and follow it, and that which is bad and change it.","There is nothing more real than what cannot be seen and there is nothing more certain than what cannot be heard.","When it is obvious that the goals cannot be reached, don't adjust the goals, adjust the action steps.","Even when walking in the company of two other men, I am bound to be able to learn from them. The good points of the one I copy, the bad points of the other I correct in myself.","The Master said, “In strolling in the company of just two other persons, I am bound to find a teacher. Identifying their strengths, I follow them, and identifying their weaknesses, I reform myself accordingly.","The demands that good people make are upon themselves;<br>Those that bad people make are upon others.","Speak the truth, do not yield to anger; give, if thou art asked for little; by these three steps thou wilt go near the gods.","The Master said, “The gentleman does not serve as a vessel.”<br>(Analects 2.12)","Benevolence is love to <i>all</i> men. It is to know <i>all</i> men.","Faithfulness and sincerity are the highest things.","If a craftsman wants to do good work, he must first sharpen his tools.","Those who cannot forgive others break the bridge over which they themselves must pass.","The gentleman harmonizes [he] without being an echo. The petty man echoes [tong] and does not harmonize.","To see what is right and not do it is the worst cowardice.","If there is righteousness in the heart, there will be beauty in the character.<br>If there is beauty in the character, there will be harmony in the home.<br>If there is harmony in the home, there will be order in the nations.<br>When there is order in the nations, there will peace in the world.","He was a man, take him for all in all, I shall not look upon his like again.","Fix your mind on truth, hold firm to virtue, rely on loving kindness, and find your recreation in the Arts.","The strength of a nation derives from the integrity of the home.","Some are born with knowledge, some derive it from study, and some acquire it only after a painful realization of their ignorance.<br><br>But the knowledge being possessed, it comes to the same thing.<br><br>Some study with a natural ease, some from a desire for advantages, and some by strenuous effort.<br>","Respectfulness, without the rules of propriety, becomes laborious bustle; carefulness, without the rules of propriety, becomes timidity; boldness, without the rules of propriety, becomes insubordination; straightforwardness, without the rules of propriety, becomes rudeness.","In his dealings with the world, the gentleman is not invariably for or against anything. He is on the side of what is moral.","Be strict with yourself but least reproachful of others and complaint is kept afar.","He Who Knows And Knows That He Knows Is A Wise Man - Follow Him;<br>He Who Knows Not And Knows Not That He Knows Not Is A Fool - Shun Him","Respect yourself and others will respect you.","Life is really simple, but men insist on making it complicated.","A superior man in dealing with the world is not for anything or against anything. He follows righteousness as the standard.","I have not seen a person who loved virtue, or one<br>who hated what was not virtuous. He who loved virtue, would esteem<br>nothing above it. He who hated what is not virtuous, would practice<br>virtue in such a way that he would not allow anything that is not<br>virtuous to approach his person.","Wealth and honor are things that all people desire, and yet unless they are acquired in the proper way I will not abide them. Poverty and disgrace are things that all people hate, and yet unless they are avoided in the proper way I will not despise them. <br><br>“If the gentleman abandons ren, how can he be worthy of that name? The gentleman does not violate ren even for the amount of time required to eat a meal. Even in times of urgency or distress, he does not depart from it.","The course (of duty), virtue, benevolence, and righteousness cannot be fully carried out without the rules of propriety; nor are training and oral lessons for the rectification of manners complete; nor can the clearing up of quarrels and discriminating in disputes be accomplished; nor can (the duties between) ruler and minister, high and low, father and son, elder brother and younger, be determined; nor can students for office and (other) learners, in serving their masters, have an attachment for them; nor can majesty and dignity be shown in assigning the different places at court, in the government of the armies, and in discharging the duties of office so as to secure the operation of the laws; nor can there be the (proper) sincerity and gravity in presenting the offerings to spiritual Beings on occasions of supplication, thanksgiving, and the various sacrifices. Therefore the superior man is respectful and reverent, assiduous in his duties and not going beyond them, retiring and yielding - thus illustrating (the principle of) propriety.","Insecurity is worse than poverty.","Consideration for others is the basis of a good life, a good society.","You cannot open a book without learning something.","Learning without thought is labor lost; thought without learning is perilous.","To be wealthy and honored in an unjust society is a disgrace.","Don't do unto others what you don't want done unto you.","Silence is a true friend who never betrays.","To give comfort to the old, to have the trust of my friends, and to have the young seeking to be near me","I was not born with wisdom. I love the ancient teachings and have worked hard to attain to their level.","The Master said, Study as though you could never catch up, [and if you did,] you would still be fearful of losing it.","He who studies but does not think is lost. He who thinks but does not study is dangerous.","The gentleman understands what is moral. The small man understands what is profitable.","Attack the evil that is within yourself, rather than attacking the evil that is in others.","Sincerity is the end and beginning of things; without sincerity there would be nothing.","To see high merit and be unable to raise it to office, to raise it but not to give such promotion precedence, is just destiny.","The noble minded act before speak, then they speak according to their actions.","We all have two lives, your second one begins when you realize you’ve only got one.","Learning without thinking is useless. Thinking without learning is dangerous.","At fifteen my mind was directed to study, and at thirty I knew where to stand.","When you meet persons of exceptional character think to stand shoulder to shoulder with them; meeting persons of little character, look inward and examine yourself.","To see the right and not to do it is cowardice.","No matter how busy you may think you are, you must find time for reading, or surrender yourself to self-chosen ignorance.","Have no friends not equal to yourself.","The gentleman understands what is right, whereas the petty man understands profit.","At fifteen I set my heart upon learning. <br>At thirty, I had planted my feet firm upon the ground. <br>At forty, I no longer suffered from perplexities. <br>At fifty, I knew what were the biddings of Heaven. <br>At sixty, I heard them with docile ear. <br>At seventy, I could follow the dictates of my own heart; for what I desired no longer overstepped the boundaries of right.","Behind every smile there's teeth.","Don’t grieve when people fail to recognize your ability. Grieve for your lack of ability instead.","If what one has to say is not better than silence, then one should keep silent.","If a man be without the virtues proper to humanity, what has he to do with the rites of propriety? If a man be without the virtues proper to humanity, what has he to do with music?","A youth is to be regarded with respect. How do we know that his future will not be equal to our present?","Success depends upon previous preparation.","Listen to all that you can – then, if you forget the deficiencies and speak about the rest cautiously, you avoid trouble. See all that you can – then, if you forget the perilous and act upon the rest cautiously, you’ll avoid regret. In speech avoid trouble, in action avoid regret – then rewards will come of themselves.","Your life is what your thoughts make it.","I hear and I forget. I see and I remember. I do and I understand.","We have two lives, and the second begins when we realize we only have one.","If I understand Change, I shall make no great mistake in Life","To be poor without murmuring is difficult. To be rich without being proud is easy.","Choose a job you love, and you will never have to work a day in your life.","Very few go astray who comport themselves with restraint.","An educated woman is a worthless woman.","When you make a promise consistent with what is right, you can keep your word. When you show respect consistent with good taste, you keep shame and disgrace at a distance. When he in whom you confide is one who does not fail his friends, you may trust him fully.","He who wishes to secure the good of others, has already secured his own.","Music produces a kind of pleasure which human nature cannot do without.","Worthy admonitions cannot fail to inspire us, but what matters is changing ourselves. Reverent advice cannot fail to encourage us, but what matters is acting on it. Encouraged without acting, inspired without changing – there’s nothing to be done for such people.","The noble-minded worry about their lack of ability, not about people’s failure to recognize their ability.","You can force the people to obey; you cannot force them to understand.","The will to win, the desire to succeed, the urge to reach your full potential... these are the keys that will unlock the door to personal excellence.","To one who is not eager I do not reveal anything, nor do I explain anything to one who is not communicative. If I raise one corner for someone and he cannot come back with the other three, I do not go on.","I followed my heart without breaking any rules.","When you see a person of worth, concentrate upon becoming their equal.","Do not do to others what you do not want them to do to you.","The Master said, 'Can men refuse to assent to the words of strict admonition? But it is reforming the conduct because of them which is valuable. Can men refuse to be pleased with words of gentle advice? But it is unfolding their aim which is valuable. If a man be pleased with these words, but does not unfold their aim, and assents to those, but does not reform his conduct, I can really do nothing with him.","Hold faithfulness and sincerity as first principles.","The superior man is modest in his speech, but exceeds in his actions.","The gentleman does not promote people merely on the basis of their words, nor does he reject words merely because of the person who uttered them.","Think of tomorrow, the past can't be mended.","To put the world in order, we must first put the nation in order; to put the nation in order, we must first put the family in order; to put the family in order; we must first cultivate our personal life; we must first set our hearts right.","Men’s natures are alike; it is their habits that separate them.","The more man meditates upon good thoughts, the better will be his world and the world at large.","If you want to make a stand, help others make a stand, and if you want to reach your goal, help others reach their goal. Consider yourself and treat others accordingly: this is the method of Humanity.","It is easy to hate and it is difficult to love. This is how the whole scheme of things works. All good things are difficult to achieve; and bad things are very easy to get.","The Essence of Knowledge is, having it, to use it.","Mankind differs from the animals only by a little and most people throw that away.","By three methods we may learn wisdom: First, by reflection, which is noblest; Second, by imitation, which is easiest; and third by experience, which is the bitterest.","If there were one word that could act as a standard of conduct for one's entire life, perhaps it would be 'thoughtfulness.","The essence of knowledge is, having it, to apply it; not having it, to confess your ignorance.","He said : Not to know the decree is to be <br>without the means of being a proper man. <br>2. Not to know the rites is to be without means to <br>construct. <br>3. Not to know words is to <br>be without the fluid needful to understand men.","He who is concerned only with the purity of his own life ruins the great human relations.","Choose a job you love and you'll never have to work a day in your life","No matter where you go, there you are.","People with virtue must speak out; People who speak are not all virtuous.","Is not reciprocity such a word? What you do not want done to yourself, do not do to others.","Worry not that no one knows you; seek to be worth knowing.","Man who fart in church, sit in pew.","An educated woman is a stupid woman","To demand much from oneself personally, and not overmuch from others, will keep ill will at a distance.","To see what’s right and not do it is cowardice.","In following the Way, the noble-minded treasure three things: a manner free of violence and arrogance, a countenance full of sincerity and trust, a voice free of vulgarity and impropriety.","I do not open up the truth to one who is not eager to get knowledge, nor help out anyone who is not anxious to explain himself. When I have presented one corner of a subject to anyone, and he cannot from it learn the other three, I do not repeat my lesson.","Life is simple, but we insist on making it complicated.","The firm, the enduring, the simple, and the modest are near to virtue.","To know what you know and what you do not know, that is true knowledge.","If the people be led by laws, and uniformity sought to be given them by punishments, they will try to avoid the punishment, but have no sense of shame. 2. 'If they be led by virtue, and uniformity sought to be given them by the rules of propriety, they will have the sense of shame, and moreover will become good.","Never tire to study. And to teach to others","Men's natures are alike, it is their habits that carry them far apart.","The wise are free from perplexities; the virtuous from anxiety; and the bold from fear.","Is it not indeed a pleasure to acquire knowledge and constantly to exercise oneself therein?","In a country well governed, poverty is something to be ashamed of. In a country badly governed, wealth is something to be ashamed of.","When the Way prevails in the world, then be seen. When it does not, then hide.","The Master said, Guide the people by law, aline them by punishment; they may shun crime, but they will want shame. Guide them by mind, aline them by courtesy; they will learn shame and grow good.","He had four of the characteristics of a superior man:— in his conduct of himself, he was humble; in serving his superiors, he was respectful; in nourishing the people, he was kind; in ordering the people, he was just.","Never contract friendship with a man that is not better than thyself. ","He who flatters a man is his enemy. he who tells him of his faults is his maker.","The noble-minded are all-encompassing, not stuck in doctrines. Little people are stuck in doctrines.","It is not the failure of others to appreciate your abilities that should trouble you, but rather your failure to appreciate theirs.","I am not concerned that I am not known; I seek to be worthy to be known.","If the search for riches were sure to be successful, though I should become a groom with a whip in my hand to get them, I will do so. As the search may not be successful, I will follow after that which I love.","those who understands is not better than those who appreciates, those who appreciates is not better than those who enjoys.","The faults of the superior man are like the eclipses of the sun and moon. He has his faults, and all men see them; he changes, and all men look up to him.","Am I a man of great wisdom? Hardly! Even when a simple person brings me a question, my mind goes utterly blank, I just thrash it out until I’ve exhausted every possibility.","Coarse rice to eat, water to drink, my bended arm for a pillow - therein is happiness. Wealth and rank attained through immoral means are nothing but drifting clouds.","I do not open the way for students who are not driven with eagerness; I do not supply a vocabulary for students who are not trying desperately to find the language for their ideas. If on showing students one corner they do not come back to me with the other three, I will not repeat myself.","We have two lives; the second begins when we realize we only have one.","Knowing it does not compare with loving it; loving it does not compare with delighting in it.","To go into battle with people who have not been properly trained is to forsake them.","I slept and dreamt life is beauty, I woke and found life is duty.","A common man marvels at uncommon things. A wise man marvels at the commonplace."]})

In [ ]:
with open(DIR+'quotes.json', 'w') as f:
    f.write(json_string)

In [ ]:
tate_probs = pd.read_csv(DIR+'tate_probs.csv').sort_values(by='probability',ascending=False).reset_index(drop=True).drop(columns=['Unnamed: 0'])

In [ ]:
tate_probs.iloc[19:120,]

quote  probability  \
19                   I have truly conquered the world.   -57.790835   
20                A real brother always has your back.   -58.236985   
21                       You said tomorrow, yesterday.   -58.857096   
22   I know of no better life purpose than to peris...   -60.205747   
23                  Life is easy for the professional.   -60.363115   
..                                                 ...          ...   
115  There is no honor in comparing yourself to los...  -104.850575   
116  I believe that a man’s life is difficult and t...  -105.412510   
117                      MPBUH https://t.co/LQDR8SbOn1  -105.532750   
118  What can my enemies do to me? \n\nMy paradise ...  -105.936771   
119       Correct. Learn here: https://t.co/P0AAPg5JIC  -107.005876   

                                              logprobs  
19   [None, -8.736711, -5.371924, -1.8163619, -10.7...  
20   [None, -8.736154, -5.3703384, -1.8178158, -10....  
21   [None, -8.736711, -5.371924, -1.8163619, -10.7...  
22   [None, -8.736942, -5.370904, -1.8167248, -10.7...  
23   [None, -8.736711, -5.371924, -1.8163619, -10.7...  
..                                                 ...  
115  [None, -8.736711, -5.371924, -1.8163619, -10.7...  
116  [None, -8.736711, -5.371924, -1.8163619, -10.7...  
117  [None, -8.736711, -5.371924, -1.8163619, -10.7...  
118  [None, -8.737378, -5.3696976, -1.8179363, -10....  
119  [None, -8.736711, -5.371924, -1.8163619, -10.7...  

[101 rows x 3 columns]

# Using Codex Model from OpenAI

I tried to use OpenAI's codex model to check the similarity, but results were not incredible so I stuck with the above model. Here's the code anyway.

In [ ]:
# get your own!!!
OPENAI_API_KEY = "SET_KEY"

In [5]:
# pip install openai

In [6]:
pip install tenacity

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [7]:
# # using codex
import os
import openai
import pandas as pd
openai.api_key = OPENAI_API_KEY

In [ ]:
def print_optional_kwargs(func):
    import inspect
    # Get the list of arguments of the input function
    args = inspect.getfullargspec(func).args
    # Get the list of default values of the input function
    defaults = inspect.getfullargspec(func).defaults
    # Check if there are any default values
    if defaults:
        # Iterate through the list of arguments and default values
        for i, arg in enumerate(args[-len(defaults):]):
            print(f"{arg} (optional)")
    else:
        print("No optional keyword arguments.")

In [ ]:
quotes_and_logprobs = {}
tate_quotes = list(pd.read_csv('tate.csv').loc[:,'full_text'])

In [ ]:
from tqdm import tqdm
import time
from tenacity import (
    retry,
    stop_after_attempt,
    wait_random_exponential,
)  # for exponential backoff

In [ ]:
@retry(wait=wait_random_exponential(min=1, max=60), stop=stop_after_attempt(6))
def completion_with_backoff(**kwargs):
    return openai.Completion.create(**kwargs)

time.sleep(120)
quotes_and_logprobs = []
count = 0
for quote in tqdm(tate_quotes):

    time.sleep(8)
    result = completion_with_backoff(
      model="code-davinci-002",
      prompt=f"This sentence sounds like Confucius: {quote}",
      max_tokens=0,
      temperature=0,
      logprobs=0,
      echo=True
    )

    probability = sum(result.choices[0].logprobs.token_logprobs[1:])
    all_logprobs = result.choices[0].logprobs.token_logprobs
    quotes_and_logprobs.append([quote, probability, all_logprobs])

# df = pd.DataFrame(quotes_and_logprobs, columns=["quote", "probability", "logprobs"])
# df.to_csv('tate_probs.csv')
# df.sort_values(by='probability',ascending=False).iloc[20:60,]